lib/SILOptimizer/Transforms/TempRValueElimination.cpp

```diff
@@ -474,33 +487,48 @@ void TempRValueOptPass::run() {
   bool changed = false;
 
   // Find all copy_addr instructions.
+  llvm::SmallPtrSet<CopyAddrInst *, 8> deadCopies;
```

Comment:
Can you use a small vector here? If you are worried about multiple dead copies, can you just do a sortUnique on a small vector?

Reply:
Yes, the same copy can be inserted multiple times, which I was trying to communicate with a set. I can change it to sortUnique.

Reply:
Also, I was originally checking membership and deleting from the set, but I'm not doing that anymore.